### DISCUSSION TITLE
evaluate configurations based on operating system

### DISCUSSION THREAD

**Discussion by FractalDesigns** [apple/pkl#408] [Q&A] [answered]
Hello, 

I am trying to create a pkl configuration that output a toml file based on the operating system. (diffrent keys are not same for each os) 
Is it possible to do it with pkl ?

I can't find anything relevant to operating system. 
if it is possible can you provide an example on how to use it for mac/linux/windows please ? 

Thanks in advance.

**Comment by odenix** [comment, marked as answer]
This should work:
```
import "pkl:platform"

os = platform.current.operatingSystem.name
```
https://pkl-lang.org/package-docs/pkl/current/platform/index.html

I'm a little surprised this exists because it can result in non-deterministic output.

**Reply by FractalDesigns** [reply]
Thank you. 
The reason i need this is to manage my dotfiles. 
One simple example in alacritty is to remove window decoration on mac but not in windows or linux (the configuration key only works on mac)
pkl lang seems to be a fit for configuring my dotfiles for my diffrent laptops and operating systems.
